appservice/src/createAppService/AppServicePlanCreateStep.ts

```diff
@@ -20,21 +20,32 @@ export class AppServicePlanCreateStep extends AzureWizardExecuteStep<IAppService
             const newPlanName: string = wizardContext.newPlanName!;
             // tslint:disable-next-line:no-non-null-assertion
             const newSku: SkuDescription = wizardContext.newPlanSku!;
-            const findingAppServicePlan: string = localize('FindingAppServicePlan', 'Ensuring App Service plan "{0}" with pricing tier "{1}" exists...', newPlanName, newSku.name);
+            const findingAppServicePlan: string = localize('FindingAppServicePlan', 'Ensuring App Service plan "{0}" exists...', newPlanName);
+            const creatingAppServicePlan: string = localize('CreatingAppServicePlan', 'Creating App Service plan "{0}"...', newPlanName);
             const foundAppServicePlan: string = localize('FoundAppServicePlan', 'Successfully found App Service plan "{0}".', newPlanName);
-            await window.withProgress({ location: ProgressLocation.Notification, title: findingAppServicePlan}, async (): Promise<void> => {
+            const createdAppServicePlan: string = localize('CreatedAppServicePlan', 'Successfully created App Service plan "{0}".', newPlanName);
+            await window.withProgress({ location: ProgressLocation.Notification, title: findingAppServicePlan }, async (): Promise<void> => {
                 ext.outputChannel.appendLine(findingAppServicePlan);
                 const client: WebSiteManagementClient = new WebSiteManagementClient(wizardContext.credentials, wizardContext.subscriptionId);
                 addExtensionUserAgent(client);
-                wizardContext.plan = await client.appServicePlans.createOrUpdate(wizardContext.resourceGroup.name, newPlanName, {
-                    appServicePlanName: newPlanName,
-                    kind: getAppServicePlanModelKind(wizardContext.newSiteKind, wizardContext.newSiteOS),
-                    sku: newSku,
-                    location: wizardContext.location.name,
-                    reserved: wizardContext.newSiteOS === WebsiteOS.linux  // The secret property - must be set to true to make it a Linux plan. Confirmed by the team who owns this API.
+                const getPlan: AppServicePlan = await client.appServicePlans.get(wizardContext.resourceGroup.name, newPlanName); // if it does't exist, evaluates as null
```

Comment:
nit: Can we rename this to `existingPlan` and change the type to `AppServicePlan | null`